docker-base/Dockerfile

```diff
@@ -4,13 +4,19 @@
 FROM alpine:3.8
 LABEL authors="Anubhav Mishra, Luke Kysow"
 
-# create atlantis user
+# We use gosu to step down from root and run as the atlantis user so we need
+# to create that user and group.
+# We add the atlantis user to the root group and make its home directory
+# owned by root so that OpenShift users can use /home/atlantis as their
+# data dir because OpenShift runs containers as a random uid that's part of
+# the root group.
 RUN addgroup atlantis && \
-    adduser -S -G atlantis atlantis
+    adduser -S -G atlantis atlantis && \
+    adduser atlantis root && \
+    chown atlantis:root /home/atlantis/ && \
+    chmod g=u /home/atlantis/
 
-ENV ATLANTIS_HOME_DIR=/home/atlantis
```

Comment:
I don't think it's a good idea to set the `HOME` env because that's set by the shell.

Reply:
I think in this case it is a good idea because atlantis expects it to be in /home/atlantis but when running with a random user this variable would not be set and thus failing when trying to create in / (the default when there is no $HOME set)

Reply:
It doesn't expect to be in `/home/atlantis` though. The code just defaults to using `~/.atlantis` for the data directory.

If you're running as a user that doesn't have write access to your home directory then you need to specify a different directory to store the data in.

Reply:
However the only two possible scenarios are either we run with the atlantis user which has the HOME=/home/atlantis or a random uid which won't have any home by default. Thus setting HOME=/home/atlantis ensures our user will always use this as its home which is a good guarantee for us as we have changed the permission on /home/atlantis to work with any random uid (setting the group permissions correctly on it)